sklearn/linear_model/_quantile.py

```diff
@@ -0,0 +1,222 @@
+
+        if not isinstance(self.fit_intercept, bool):
+            raise ValueError(f"The argument fit_intercept must be bool, "
+                             f"got {self.fit_intercept}")
+
+        if self.solver not in (
+            "highs-ds", "highs-ipm", "highs", "interior-point",
+            "revised simplex",
+        ):
+            raise ValueError(f"Invalid value for argument solver, "
+                             f"got {self.solver}")
+        elif (
+            self.solver == "revised simplex"
+            and sp_version < parse_version('1.3.0')
+        ):
+            raise ValueError(f"Solver 'revised simplex' is only available "
+                             f"with scipy>=1.3.0, got {sp_version}")
+        elif (
+            self.solver in ("highs-ds", "highs-ipm", "highs")
+            and sp_version < parse_version('1.6.0')
+        ):
+            raise ValueError(f"Solver {self.solver} is only available "
+                             f"with scipy>=1.6.0, got {sp_version}")
+
+        # Use linear programming formulation of quantile regression
+        #     min_x c x
+        #           A_eq x = b_eq
+        #                0 <= x
+        # x = (s0, s, t0, t, u, v) = slack variables
+        # intercept = s0 + t0
+        # coef = s + t
+        # c = (alpha * 1_p, alpha * 1_p, quantile * 1_n, (1-quantile) * 1_n)
+        # residual = y - X@coef - intercept = u - v
+        # A_eq = (1_n, X, -1_n, -X, diag(1_n), -diag(1_n))
+        # b_eq = y
+        # p = n_features + fit_intercept
+        # n = n_samples
+        # 1_n = vector of length n with entries equal one
+        # see https://stats.stackexchange.com/questions/384909/
+        c = np.concatenate([
+            np.ones(n_params * 2) * alpha,
+            sample_weight * self.quantile,
+            sample_weight * (1 - self.quantile),
+        ])
+        if self.fit_intercept:
+            # do not penalize the intercept
+            c[0] = 0
+            c[n_params] = 0
+
+            A_eq = np.concatenate([
+                np.ones((n_samples, 1)),
+                X,
+                -np.ones((n_samples, 1)),
+                -X,
+                np.eye(n_samples),
+                -np.eye(n_samples),
+            ], axis=1)
+        else:
+            A_eq = np.concatenate([
+                X,
+                -X,
+                np.eye(n_samples),
+                -np.eye(n_samples),
+            ], axis=1)
+
+        b_eq = y
+
+        result = linprog(
+            c=c,
+            A_eq=A_eq,
+            b_eq=b_eq,
+            method=self.solver,
+            options=self.solver_options,
+        )
+        solution = result.x
+        if not result.success:
+            warnings.warn(
+                'Linear programming for Quantile regression did not converge. '
+                'Status is {}'.format(result.status), ConvergenceWarning
+            )
+            if solution is np.nan:
+                solution = np.zeros(A_eq.shape[1])
```

Comment:
`enet_coordinate_descent` warns instead of throwing an error. `Ridge` relies on linalg routines that, like `solve`, either warn or error. `HuberRegressor` raises a `ValueError`.
Conclusion: I'll change to raise a ValueError.

Reply:
I added a test for failing linprog, too.